### PR TITLE
Add atanxy(x, y)

### PR DESCRIPTION
**Pull request by arseniiv**
The standing, and likely very archaic, convention of `atan2` having to be called in all major contexts with arguments `(y, x)` swapped around, is probably instructed by that it’s “like `atan(y / x)` but more correct”. This reasoning is not very good, *especially* when languages call these parameters `x` and `y` in their direct order in the docs. People are bound to be confused or, when sophisticated enough, being anxious about if the convention is the usual reversed one or a revised one, and taing time to look at the docs.

What can be made instead in the face of this convention’s existence is having a function with another name like `atanxy` which dictates a bit more clearly that its arguments are most probably coordinates and they go in their usual order. So here is a patch adding this option to SW3. IMO it befits its cutting-edge (and a bit whimsical!) nature.

### CODE REVIEW

**Comment by frostburn** [comment]
I think the other atan should be called `atanyx` then and the params renamed. Also we're in `camelCase` land so maybe `atanXY` and `atanYX` instead.
Feel free to expand the docstrings to explain what the two argument atan is for in the first place.

**Comment by arseniiv** [comment]
If we’re renaming `atan2` then it can be just removed altogether? I thought it should be left under its standard name for compatibility.

Made `atanXY`’s doc be
> Calculate atanXY(x, y) = atan2(y, x) which is the angle between (1, 0) and (x, y), chosen to lie in (−π; π], positive anticlockwise.

I’m not changing anything about `atan2` yet (except parameter names and doc: done) without understanding better what’s the plan with its support. 🤔🤔🤔

File: src/builtin.ts [hunk 1]
```diff
@@ -138,15 +138,26 @@ for (const name of MATH_KEYS) {
...
   );
 }
-atan2.__doc__ = 'Calculate atan2(x, y).';
+atan2.__doc__ = 'Calculate atan2(y, x) which is the angle between (1, 0) and (x, y), chosen to lie in (−π; π], positive anticlockwise.';
 atan2.__node__ = builtinNode(atan2);
 
+// Equivalent to atan2 but with swapped arguments.
+// Rationale is that atanXY(x, y) = log(x + i * y), x and y now coordinates.
+function atanXY(x: Interval, y: Interval) {
+  return new Interval(
```
Comment (frostburn) [inline src/builtin.ts:153]: Just call `atan2` defined above with swapped arguments. Leaves less room for error if there's more domain finagling in the future or something.

**Comment by frostburn** [comment]
Looks good now besides the minor comment about code re-use and the missing re-generated auto-docs. (If you don't have `node` or `bun` it's fine to leave them out.)

File: src/builtin.ts (hunk 1 above)
Comment (arseniiv) [reply src/builtin.ts:153]: Yeah I don’t know why I left it that way. Ah I remember now, because I wasn’t sure if `atan2` is to live.

**Comment by arseniiv** [comment]
> This branch has conflicts that must be resolved

Good work me. 😂

**Comment by frostburn** [comment]
`main` is a moving target. :sweat_smile: How about I just re-implement this on my side and assign you as co-author in the commit?

**Comment by arseniiv** [comment]
Agree.

**Comment by frostburn** [comment]
Done.

**Pull request closed without merging**